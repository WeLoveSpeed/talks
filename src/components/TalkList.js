import React, { Component } from 'react';
import {List, ListItem, makeSelectable} from 'material-ui/List';
import Avatar from 'material-ui/Avatar';
import {teal500, red400, orange500} from 'material-ui/styles/colors';
import {Tabs, Tab} from 'material-ui/Tabs';
import {isPK, isLT} from '../selectors/Talks';
import SelectField from 'material-ui/SelectField';
import MenuItem from 'material-ui/MenuItem';

let SelectableList = makeSelectable(List);

function wrapState(ComposedComponent) {
  return class SelectableList extends Component {
    componentWillMount() {
      this.setState({
        selectedIndex: this.props.defaultValue,
      });
    }

    componentWillReceiveProps(nextProps) {
      if (nextProps.defaultValue !== undefined && nextProps.defaultValue !== this.state.selectedIndex) {
        this.setState({
          selectedIndex: nextProps.defaultValue
        });
      }
    }

    handleRequestChange = (event, index) => {
      this.setState({
        selectedIndex: index,
      });
    };

    render() {
      return (
        <ComposedComponent
          value={this.state.selectedIndex}
          onChange={this.handleRequestChange}
        >
          {this.props.children}
        </ComposedComponent>
      );
    }
  };
}

SelectableList = wrapState(SelectableList);

class TalkList extends Component {
  getFormat(format) {
    if (isPK(format)) {
      return <Avatar backgroundColor={teal500}>PK</Avatar>
    }
    if (isLT(format)) {
      return <Avatar backgroundColor={orange500}>LT</Avatar>
    }

    return null;
  }

  getTalk(talk, i) {
    return (
        <ListItem
            key={i}
            value={i}
            onTouchTap={() => this.props.selectTalk(talk.id)}
            primaryText={talk.titre_de_ta_presentation}
            secondaryText={talk.prenom_et_nom}
            rightIcon={<span>{Math.round(talk.note)}</span>}
            leftAvatar={this.getFormat(talk.formats)}
          />
    )
  }

  render() {
    const {talks, count} = this.props;
    return (
      <div className="TalkList">
        <Tabs 
          onChange={value => this.props.setFilter(value)} 
          tabItemContainerStyle={{backgroundColor: red400}}
          inkBarStyle={{backgroundColor: 'white'}}>
          <Tab label={`All (${count.all})`} value={null} />
          <Tab label={`PK (${count.PK})`} value="PK" />
          <Tab label={`LT (${count.LT})`} value="LT" />
        </Tabs>
        <div style={{padding: '0 16px'}}>
          <SelectField
            style={{width: '100%'}}
            floatingLabelText="Trier par"
            value={this.props.sortBy}
            onChange={(event, index, value) => this.props.sortTalks(value)}
          >
            <MenuItem value="date" primaryText="Date" />
            <MenuItem value="note" primaryText="Note" />
          </SelectField>
        </div>
        <SelectableList>
          {talks.map((talk, i) => this.getTalk(talk, i))}
        </SelectableList>
      </div>
    );
  }
}

export default TalkList;
