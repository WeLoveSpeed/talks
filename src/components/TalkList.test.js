import React, { Component } from 'react';
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer';
import TalkList from './TalkList';
import SelectField from 'material-ui/SelectField';
import {Tabs} from 'material-ui/Tabs';
import {ListItem} from 'material-ui/List';

const talks = [
  { 
    titre_de_ta_presentation: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dolores, esse.',
    prenom_et_nom: 'Alexandra Janin', 
    formats: 'Lightning Talk : 5 minutes',
  },
  { 
    titre_de_ta_presentation: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dolores, esse.',
    prenom_et_nom: 'Alexandra Janin', 
    formats: 'Pecha Kucha : 20 images x 20 secondes',
  },
  { 
    titre_de_ta_presentation: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dolores, esse.',
    prenom_et_nom: 'Alexandra Janin'
  }
];

describe('TalkList component', () => {
  it('should renders correctly', () => {
    const sortTalks = jest.fn(()=>Promise.resolve(0))
    const setFilter = jest.fn(()=>Promise.resolve(0))
    const selectTalk = jest.fn(()=>Promise.resolve(0))

    const component = shallow(
        <TalkList 
            selectedTalk={0}
            count={{'all': 3, 'PK': 2, 'PK': 1}}
            talks={talks}
            selectTalk={id => selectTalk(id)}
            setFilter={format => setFilter(format)}
            // sortBy={this.props.sortBy}
            sortTalks={value => sortTalks(value)} 
            />
    );
    component.find(Tabs).first().simulate('change')
    component.find(SelectField).first().simulate('change')
    component.find(ListItem).first().simulate('touchTap')
    expect(component).toMatchSnapshot();
  });
});