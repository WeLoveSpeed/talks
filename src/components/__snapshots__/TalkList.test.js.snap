exports[`TalkList component should renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="TalkList">
    <Tabs
        initialSelectedIndex={0}
        inkBarStyle={
            Object {
                "backgroundColor": "white",
              }
        }
        onChange={[Function]}
        tabItemContainerStyle={
            Object {
                "backgroundColor": "#ef5350",
              }
        }>
        <Tab
            label="All (3)"
            value={null} />
        <Tab
            label="PK (1)"
            value="PK" />
        <Tab
            label="LT (undefined)"
            value="LT" />
    </Tabs>
    <div
        style={
            Object {
                "padding": "0 16px",
              }
        }>
        <SelectField
            autoWidth={false}
            disabled={false}
            floatingLabelText="Trier par"
            fullWidth={false}
            onChange={[Function]}
            style={
                Object {
                    "width": "100%",
                  }
            }
            value={undefined}>
            <MenuItem
                checked={false}
                desktop={false}
                disabled={false}
                focusState="none"
                insetChildren={false}
                primaryText="Date"
                value="date" />
            <MenuItem
                checked={false}
                desktop={false}
                disabled={false}
                focusState="none"
                insetChildren={false}
                primaryText="Note"
                value="note" />
        </SelectField>
    </div>
    <SelectableList>
        <ListItem
            autoGenerateNestedIndicator={true}
            disableKeyboardFocus={false}
            disabled={false}
            initiallyOpen={false}
            insetChildren={false}
            leftAvatar={
                <Avatar
                    backgroundColor="#ff9800"
                    size={40}>
                    LT
                </Avatar>
            }
            nestedItems={Array []}
            nestedLevel={0}
            onKeyboardFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onNestedListToggle={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            onTouchTap={[Function]}
            open={null}
            primaryText="Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dolores, esse."
            primaryTogglesNestedList={false}
            rightIcon={<span />}
            secondaryText="Alexandra Janin"
            secondaryTextLines={1}
            value={0} />
        <ListItem
            autoGenerateNestedIndicator={true}
            disableKeyboardFocus={false}
            disabled={false}
            initiallyOpen={false}
            insetChildren={false}
            leftAvatar={
                <Avatar
                    backgroundColor="#009688"
                    size={40}>
                    PK
                </Avatar>
            }
            nestedItems={Array []}
            nestedLevel={0}
            onKeyboardFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onNestedListToggle={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            onTouchTap={[Function]}
            open={null}
            primaryText="Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dolores, esse."
            primaryTogglesNestedList={false}
            rightIcon={<span />}
            secondaryText="Alexandra Janin"
            secondaryTextLines={1}
            value={1} />
        <ListItem
            autoGenerateNestedIndicator={true}
            disableKeyboardFocus={false}
            disabled={false}
            initiallyOpen={false}
            insetChildren={false}
            leftAvatar={null}
            nestedItems={Array []}
            nestedLevel={0}
            onKeyboardFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onNestedListToggle={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            onTouchTap={[Function]}
            open={null}
            primaryText="Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dolores, esse."
            primaryTogglesNestedList={false}
            rightIcon={<span />}
            secondaryText="Alexandra Janin"
            secondaryTextLines={1}
            value={2} />
    </SelectableList>
</div>,
  "nodes": Array [
    <div
      className="TalkList">
      <Tabs
            initialSelectedIndex={0}
            inkBarStyle={
                  Object {
                        "backgroundColor": "white",
                      }
            }
            onChange={[Function]}
            tabItemContainerStyle={
                  Object {
                        "backgroundColor": "#ef5350",
                      }
            }>
            <Tab
                  label="All (3)"
                  value={null} />
            <Tab
                  label="PK (1)"
                  value="PK" />
            <Tab
                  label="LT (undefined)"
                  value="LT" />
      </Tabs>
      <div
            style={
                  Object {
                        "padding": "0 16px",
                      }
            }>
            <SelectField
                  autoWidth={false}
                  disabled={false}
                  floatingLabelText="Trier par"
                  fullWidth={false}
                  onChange={[Function]}
                  style={
                        Object {
                              "width": "100%",
                            }
                  }
                  value={undefined}>
                  <MenuItem
                        checked={false}
                        desktop={false}
                        disabled={false}
                        focusState="none"
                        insetChildren={false}
                        primaryText="Date"
                        value="date" />
                  <MenuItem
                        checked={false}
                        desktop={false}
                        disabled={false}
                        focusState="none"
                        insetChildren={false}
                        primaryText="Note"
                        value="note" />
            </SelectField>
      </div>
      <SelectableList>
            <ListItem
                  autoGenerateNestedIndicator={true}
                  disableKeyboardFocus={false}
                  disabled={false}
                  initiallyOpen={false}
                  insetChildren={false}
                  leftAvatar={
                        <Avatar
                              backgroundColor="#ff9800"
                              size={40}>
                              LT
                        </Avatar>
                  }
                  nestedItems={Array []}
                  nestedLevel={0}
                  onKeyboardFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onNestedListToggle={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                  onTouchTap={[Function]}
                  open={null}
                  primaryText="Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dolores, esse."
                  primaryTogglesNestedList={false}
                  rightIcon={<span />}
                  secondaryText="Alexandra Janin"
                  secondaryTextLines={1}
                  value={0} />
            <ListItem
                  autoGenerateNestedIndicator={true}
                  disableKeyboardFocus={false}
                  disabled={false}
                  initiallyOpen={false}
                  insetChildren={false}
                  leftAvatar={
                        <Avatar
                              backgroundColor="#009688"
                              size={40}>
                              PK
                        </Avatar>
                  }
                  nestedItems={Array []}
                  nestedLevel={0}
                  onKeyboardFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onNestedListToggle={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                  onTouchTap={[Function]}
                  open={null}
                  primaryText="Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dolores, esse."
                  primaryTogglesNestedList={false}
                  rightIcon={<span />}
                  secondaryText="Alexandra Janin"
                  secondaryTextLines={1}
                  value={1} />
            <ListItem
                  autoGenerateNestedIndicator={true}
                  disableKeyboardFocus={false}
                  disabled={false}
                  initiallyOpen={false}
                  insetChildren={false}
                  leftAvatar={null}
                  nestedItems={Array []}
                  nestedLevel={0}
                  onKeyboardFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onNestedListToggle={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                  onTouchTap={[Function]}
                  open={null}
                  primaryText="Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dolores, esse."
                  primaryTogglesNestedList={false}
                  rightIcon={<span />}
                  secondaryText="Alexandra Janin"
                  secondaryTextLines={1}
                  value={2} />
      </SelectableList>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <TalkList
        count={
                Object {
                        "PK": 1,
                        "all": 3,
                      }
        }
        notes={
                Array [
                        Object {
                          "total": 1,
                          "values": Object {
                            "A": 1,
                            "B": 0,
                            "C": 0,
                          },
                        },
                        Object {
                          "total": 1,
                          "values": Object {
                            "A": 1,
                            "B": 0,
                            "C": 0,
                          },
                        },
                        Object {
                          "total": 0,
                          "values": Object {
                            "A": 0,
                            "B": 0,
                            "C": 0,
                          },
                        },
                      ]
        }
        selectTalk={[Function]}
        selectedTalk={0}
        setFilter={[Function]}
        sortTalks={[Function]}
        talks={
                Array [
                        Object {
                          "formats": "Lightning Talk : 5 minutes",
                          "prenom_et_nom": "Alexandra Janin",
                          "titre_de_ta_presentation": "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dolores, esse.",
                        },
                        Object {
                          "formats": "Pecha Kucha : 20 images x 20 secondes",
                          "prenom_et_nom": "Alexandra Janin",
                          "titre_de_ta_presentation": "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dolores, esse.",
                        },
                        Object {
                          "prenom_et_nom": "Alexandra Janin",
                          "titre_de_ta_presentation": "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dolores, esse.",
                        },
                      ]
        } />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": TalkList {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "count": Object {
            "PK": 1,
            "all": 3,
          },
          "notes": Array [
            Object {
              "total": 1,
              "values": Object {
                "A": 1,
                "B": 0,
                "C": 0,
              },
            },
            Object {
              "total": 1,
              "values": Object {
                "A": 1,
                "B": 0,
                "C": 0,
              },
            },
            Object {
              "total": 0,
              "values": Object {
                "A": 0,
                "B": 0,
                "C": 0,
              },
            },
          ],
          "selectTalk": [Function],
          "selectedTalk": 0,
          "setFilter": [Function],
          "sortTalks": [Function],
          "talks": Array [
            Object {
              "formats": "Lightning Talk : 5 minutes",
              "prenom_et_nom": "Alexandra Janin",
              "titre_de_ta_presentation": "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dolores, esse.",
            },
            Object {
              "formats": "Pecha Kucha : 20 images x 20 secondes",
              "prenom_et_nom": "Alexandra Janin",
              "titre_de_ta_presentation": "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dolores, esse.",
            },
            Object {
              "prenom_et_nom": "Alexandra Janin",
              "titre_de_ta_presentation": "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dolores, esse.",
            },
          ],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="TalkList">
          <Tabs
                    initialSelectedIndex={0}
                    inkBarStyle={
                              Object {
                                        "backgroundColor": "white",
                                      }
                    }
                    onChange={[Function]}
                    tabItemContainerStyle={
                              Object {
                                        "backgroundColor": "#ef5350",
                                      }
                    }>
                    <Tab
                              label="All (3)"
                              value={null} />
                    <Tab
                              label="PK (1)"
                              value="PK" />
                    <Tab
                              label="LT (undefined)"
                              value="LT" />
          </Tabs>
          <div
                    style={
                              Object {
                                        "padding": "0 16px",
                                      }
                    }>
                    <SelectField
                              autoWidth={false}
                              disabled={false}
                              floatingLabelText="Trier par"
                              fullWidth={false}
                              onChange={[Function]}
                              style={
                                        Object {
                                                  "width": "100%",
                                                }
                              }
                              value={undefined}>
                              <MenuItem
                                        checked={false}
                                        desktop={false}
                                        disabled={false}
                                        focusState="none"
                                        insetChildren={false}
                                        primaryText="Date"
                                        value="date" />
                              <MenuItem
                                        checked={false}
                                        desktop={false}
                                        disabled={false}
                                        focusState="none"
                                        insetChildren={false}
                                        primaryText="Note"
                                        value="note" />
                    </SelectField>
          </div>
          <SelectableList>
                    <ListItem
                              autoGenerateNestedIndicator={true}
                              disableKeyboardFocus={false}
                              disabled={false}
                              initiallyOpen={false}
                              insetChildren={false}
                              leftAvatar={
                                        <Avatar
                                                  backgroundColor="#ff9800"
                                                  size={40}>
                                                  LT
                                        </Avatar>
                              }
                              nestedItems={Array []}
                              nestedLevel={0}
                              onKeyboardFocus={[Function]}
                              onMouseEnter={[Function]}
                              onMouseLeave={[Function]}
                              onNestedListToggle={[Function]}
                              onTouchEnd={[Function]}
                              onTouchStart={[Function]}
                              onTouchTap={[Function]}
                              open={null}
                              primaryText="Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dolores, esse."
                              primaryTogglesNestedList={false}
                              rightIcon={<span />}
                              secondaryText="Alexandra Janin"
                              secondaryTextLines={1}
                              value={0} />
                    <ListItem
                              autoGenerateNestedIndicator={true}
                              disableKeyboardFocus={false}
                              disabled={false}
                              initiallyOpen={false}
                              insetChildren={false}
                              leftAvatar={
                                        <Avatar
                                                  backgroundColor="#009688"
                                                  size={40}>
                                                  PK
                                        </Avatar>
                              }
                              nestedItems={Array []}
                              nestedLevel={0}
                              onKeyboardFocus={[Function]}
                              onMouseEnter={[Function]}
                              onMouseLeave={[Function]}
                              onNestedListToggle={[Function]}
                              onTouchEnd={[Function]}
                              onTouchStart={[Function]}
                              onTouchTap={[Function]}
                              open={null}
                              primaryText="Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dolores, esse."
                              primaryTogglesNestedList={false}
                              rightIcon={<span />}
                              secondaryText="Alexandra Janin"
                              secondaryTextLines={1}
                              value={1} />
                    <ListItem
                              autoGenerateNestedIndicator={true}
                              disableKeyboardFocus={false}
                              disabled={false}
                              initiallyOpen={false}
                              insetChildren={false}
                              leftAvatar={null}
                              nestedItems={Array []}
                              nestedLevel={0}
                              onKeyboardFocus={[Function]}
                              onMouseEnter={[Function]}
                              onMouseLeave={[Function]}
                              onNestedListToggle={[Function]}
                              onTouchEnd={[Function]}
                              onTouchStart={[Function]}
                              onTouchTap={[Function]}
                              open={null}
                              primaryText="Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dolores, esse."
                              primaryTogglesNestedList={false}
                              rightIcon={<span />}
                              secondaryText="Alexandra Janin"
                              secondaryTextLines={1}
                              value={2} />
          </SelectableList>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="TalkList">
          <Tabs
                    initialSelectedIndex={0}
                    inkBarStyle={
                              Object {
                                        "backgroundColor": "white",
                                      }
                    }
                    onChange={[Function]}
                    tabItemContainerStyle={
                              Object {
                                        "backgroundColor": "#ef5350",
                                      }
                    }>
                    <Tab
                              label="All (3)"
                              value={null} />
                    <Tab
                              label="PK (1)"
                              value="PK" />
                    <Tab
                              label="LT (undefined)"
                              value="LT" />
          </Tabs>
          <div
                    style={
                              Object {
                                        "padding": "0 16px",
                                      }
                    }>
                    <SelectField
                              autoWidth={false}
                              disabled={false}
                              floatingLabelText="Trier par"
                              fullWidth={false}
                              onChange={[Function]}
                              style={
                                        Object {
                                                  "width": "100%",
                                                }
                              }
                              value={undefined}>
                              <MenuItem
                                        checked={false}
                                        desktop={false}
                                        disabled={false}
                                        focusState="none"
                                        insetChildren={false}
                                        primaryText="Date"
                                        value="date" />
                              <MenuItem
                                        checked={false}
                                        desktop={false}
                                        disabled={false}
                                        focusState="none"
                                        insetChildren={false}
                                        primaryText="Note"
                                        value="note" />
                    </SelectField>
          </div>
          <SelectableList>
                    <ListItem
                              autoGenerateNestedIndicator={true}
                              disableKeyboardFocus={false}
                              disabled={false}
                              initiallyOpen={false}
                              insetChildren={false}
                              leftAvatar={
                                        <Avatar
                                                  backgroundColor="#ff9800"
                                                  size={40}>
                                                  LT
                                        </Avatar>
                              }
                              nestedItems={Array []}
                              nestedLevel={0}
                              onKeyboardFocus={[Function]}
                              onMouseEnter={[Function]}
                              onMouseLeave={[Function]}
                              onNestedListToggle={[Function]}
                              onTouchEnd={[Function]}
                              onTouchStart={[Function]}
                              onTouchTap={[Function]}
                              open={null}
                              primaryText="Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dolores, esse."
                              primaryTogglesNestedList={false}
                              rightIcon={<span />}
                              secondaryText="Alexandra Janin"
                              secondaryTextLines={1}
                              value={0} />
                    <ListItem
                              autoGenerateNestedIndicator={true}
                              disableKeyboardFocus={false}
                              disabled={false}
                              initiallyOpen={false}
                              insetChildren={false}
                              leftAvatar={
                                        <Avatar
                                                  backgroundColor="#009688"
                                                  size={40}>
                                                  PK
                                        </Avatar>
                              }
                              nestedItems={Array []}
                              nestedLevel={0}
                              onKeyboardFocus={[Function]}
                              onMouseEnter={[Function]}
                              onMouseLeave={[Function]}
                              onNestedListToggle={[Function]}
                              onTouchEnd={[Function]}
                              onTouchStart={[Function]}
                              onTouchTap={[Function]}
                              open={null}
                              primaryText="Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dolores, esse."
                              primaryTogglesNestedList={false}
                              rightIcon={<span />}
                              secondaryText="Alexandra Janin"
                              secondaryTextLines={1}
                              value={1} />
                    <ListItem
                              autoGenerateNestedIndicator={true}
                              disableKeyboardFocus={false}
                              disabled={false}
                              initiallyOpen={false}
                              insetChildren={false}
                              leftAvatar={null}
                              nestedItems={Array []}
                              nestedLevel={0}
                              onKeyboardFocus={[Function]}
                              onMouseEnter={[Function]}
                              onMouseLeave={[Function]}
                              onNestedListToggle={[Function]}
                              onTouchEnd={[Function]}
                              onTouchStart={[Function]}
                              onTouchTap={[Function]}
                              open={null}
                              primaryText="Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dolores, esse."
                              primaryTogglesNestedList={false}
                              rightIcon={<span />}
                              secondaryText="Alexandra Janin"
                              secondaryTextLines={1}
                              value={2} />
          </SelectableList>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <TalkList
    count={
        Object {
            "PK": 1,
            "all": 3,
          }
    }
    notes={
        Array [
            Object {
              "total": 1,
              "values": Object {
                "A": 1,
                "B": 0,
                "C": 0,
              },
            },
            Object {
              "total": 1,
              "values": Object {
                "A": 1,
                "B": 0,
                "C": 0,
              },
            },
            Object {
              "total": 0,
              "values": Object {
                "A": 0,
                "B": 0,
                "C": 0,
              },
            },
          ]
    }
    selectTalk={[Function]}
    selectedTalk={0}
    setFilter={[Function]}
    sortTalks={[Function]}
    talks={
        Array [
            Object {
              "formats": "Lightning Talk : 5 minutes",
              "prenom_et_nom": "Alexandra Janin",
              "titre_de_ta_presentation": "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dolores, esse.",
            },
            Object {
              "formats": "Pecha Kucha : 20 images x 20 secondes",
              "prenom_et_nom": "Alexandra Janin",
              "titre_de_ta_presentation": "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dolores, esse.",
            },
            Object {
              "prenom_et_nom": "Alexandra Janin",
              "titre_de_ta_presentation": "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dolores, esse.",
            },
          ]
    } />,
}
`;
